File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/CatalogPageRow.java
package io.trino.tpcds.row;

import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_CATALOG_NUMBER;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_CATALOG_PAGE_ID;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_CATALOG_PAGE_NUMBER;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_CATALOG_PAGE_SK;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_DEPARTMENT;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_DESCRIPTION;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_END_DATE_ID;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_START_DATE_ID;
import static io.trino.tpcds.generator.CatalogPageGeneratorColumn.CP_TYPE;

import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

public class CatalogPageRow extends KyuubiTPCDSTableRowWithNulls {
  private final long cpCatalogPageSk;
  private final String cpCatalogPageId;
  private final long cpStartDateId;
  private final long cpEndDateId;
  private final String cpDepartment;
  private final int cpCatalogNumber;
  private final int cpCatalogPageNumber;
  private final String cpDescription;
  private final String cpType;

  public CatalogPageRow(
      long cpCatalogPageSk,
      String cpCatalogPageId,
      long cpStartDateId,
      long cpEndDateId,
      String cpDepartment,
      int cpCatalogNumber,
      int cpCatalogPageNumber,
      String cpDescription,
      String cpType,
      long nullBitMap) {
    super(nullBitMap, CP_CATALOG_PAGE_SK);
    this.cpCatalogPageSk = cpCatalogPageSk;
    this.cpCatalogPageId = cpCatalogPageId;
    this.cpStartDateId = cpStartDateId;
    this.cpEndDateId = cpEndDateId;
    this.cpDepartment = cpDepartment;
    this.cpCatalogNumber = cpCatalogNumber;
    this.cpCatalogPageNumber = cpCatalogPageNumber;
    this.cpDescription = cpDescription;
    this.cpType = cpType;
  }

  public long getCpCatalogPageSk() {
    return cpCatalogPageSk;
  }

  public String getCpCatalogPageId() {
    return cpCatalogPageId;
  }

  public long getCpStartDateId() {
    return cpStartDateId;
  }

  public long getCpEndDateId() {
    return cpEndDateId;
  }

  public String getCpDepartment() {
    return cpDepartment;
  }

  public int getCpCatalogNumber() {
    return cpCatalogNumber;
  }

  public int getCpCatalogPageNumber() {
    return cpCatalogPageNumber;
  }

  public String getCpDescription() {
    return cpDescription;
  }

  public String getCpType() {
    return cpType;
  }

  @Override
  public Object[] values() {
    return new Object[] {
      getOrNullForKey(cpCatalogPageSk, CP_CATALOG_PAGE_SK),
      getOrNull(cpCatalogPageId, CP_CATALOG_PAGE_ID),
      getOrNullForKey(cpStartDateId, CP_START_DATE_ID),
      getOrNullForKey(cpEndDateId, CP_END_DATE_ID),
      getOrNull(cpDepartment, CP_DEPARTMENT),
      getOrNull(cpCatalogNumber, CP_CATALOG_NUMBER),
      getOrNull(cpCatalogPageNumber, CP_CATALOG_PAGE_NUMBER),
      getOrNull(cpDescription, CP_DESCRIPTION),
      getOrNull(cpType, CP_TYPE)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/CatalogPageRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.trino.tpcds.row

import io.trino.tpcds.generator.CatalogPageGeneratorColumn._
import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
import org.apache.spark.sql.catalyst.InternalRow

import java.util.{List => JList}

class CatalogPageRow(
   private val cpCatalogPageSk: Long,
   private val cpCatalogPageId: String,
   private val cpStartDateId: Long,
   private val cpEndDateId: Long,
   private val cpDepartment: String,
   private val cpCatalogNumber: Int,
   private val cpCatalogPageNumber: Int,
   private val cpDescription: String,
   private val cpType: String,
   nullBitMap: Long) extends TableRowWithNulls(nullBitMap, CP_CATALOG_PAGE_SK)
  with KyuubiTableRowWithNulls {

  override val nullBitMapInternal = nullBitMap
  override val firstColumnInternal = CP_CATALOG_PAGE_SK

  def getCpCatalogPageSk: Long = cpCatalogPageSk
  def getCpCatalogPageId: String = cpCatalogPageId
  def getCpStartDateId: Long = cpStartDateId
  def getCpEndDateId: Long = cpEndDateId
  def getCpDepartment: String = cpDepartment
  def getCpCatalogNumber: Int = cpCatalogNumber
  def getCpCatalogPageNumber: Int = cpCatalogPageNumber
  def getCpDescription: String = cpDescription
  def getCpType: String = cpType

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
        getLongOrNull(cpCatalogPageSk, CP_CATALOG_PAGE_SK),
        getStringOrNullInternal(cpCatalogPageId, CP_CATALOG_PAGE_ID),
        getIntOrNull(cpStartDateId, CP_START_DATE_ID),
        getIntOrNull(cpEndDateId, CP_END_DATE_ID),
        getStringOrNullInternal(cpDepartment, CP_DEPARTMENT),
        getIntOrNull(cpCatalogNumber, CP_CATALOG_NUMBER),
        getIntOrNull(cpCatalogPageNumber, CP_CATALOG_PAGE_NUMBER),
        getStringOrNullInternal(cpDescription, CP_DESCRIPTION),
        getStringOrNullInternal(cpType, CP_TYPE)))
}
